.github/workflows/template.flink-ci.yml

```diff
@@ -0,0 +1,438 @@
+
+      - name: "Load Docker images if not present in cache, yet"
+        if: ${{ !cancelled() && !steps.docker-cache.cache.hit }}
+        working-directory: ${{ env.CONTAINER_LOCAL_WORKING_DIR }}
+        run: ./tools/azure-pipelines/cache_docker_images.sh load
+
+      - name: "Test - ${{ matrix.module }}"
+        id: test-run
+        working-directory: ${{ env.CONTAINER_LOCAL_WORKING_DIR }}
+        env:
+          IT_CASE_S3_BUCKET: ${{ secrets.s3_bucket }}
+          IT_CASE_S3_ACCESS_KEY: ${{ secrets.s3_access_key }}
+          IT_CASE_S3_SECRET_KEY: ${{ secrets.s3_secret_key }}
+        timeout-minutes: 240
+        run: |
+          ${{ inputs.environment }} PROFILE="$PROFILE -Pgithub-actions" ./tools/azure-pipelines/uploading_watchdog.sh \
+              -a ${{ github.job }} \
+              -d ${{ env.FLINK_ARTIFACT_DIR }} \
+              -t 480 \
+              ./tools/ci/test_controller.sh ${{ matrix.module }}
+
+      - name: "Post-process build artifacts"
+        working-directory: ${{ env.CONTAINER_LOCAL_WORKING_DIR }}
+        run: find ${{ steps.test-run.outputs.debug-files-output-dir }} -type f -exec rename 's/[:<>|*?]/-/' {} \;
+
+      - name: "Upload build artifacts"
+        uses: actions/upload-artifact@v3
+        if: ${{ failure() && steps.test-run.outputs.debug-files-output-dir }} != ''
+        with:
+          name: logs-test-${{ needs.compile.outputs.stringified-workflow-name }}-${{ github.run_number }}-${{ matrix.stringified-module-name }}-${{ steps.test-run.outputs.debug-files-name }}
+          path: ${{ steps.test-run.outputs.debug-files-output-dir }}
+
+      - name: "Save Docker images to cache"
+        working-directory: ${{ env.CONTAINER_LOCAL_WORKING_DIR }}
+        if: ${{ !cancelled() && (failure() || !steps.docker-cache.cache.hit) }}
+        run: ./tools/azure-pipelines/cache_docker_images.sh save
+
+  e2e-prereq-check:
+    name: "Check: Code modified"
+    needs: compile
+    runs-on: ubuntu-latest
+    container:
+      image: chesnay/flink-ci:java_8_11_17_21_maven_386
+      options: --init
+    outputs:
+      skip-e2e: ${{ steps.docs-only-pr-check.skip-e2e }}
+    steps:
+      - name: "Flink Checkout"
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+          persist-credentials: false
+
+      # Skip e2e test execution if this is a documentation only pull request (master / release builds will still be checked regularly)
+      - name: "Check if it's a docs-only PR (i.e. e2e tests can be skipped)"
+        id: docs-only-pr-check
+        run: |
+          source ./tools/azure-pipelines/build_properties.sh
+          if is_docs_only_pullrequest; then
+            echo "This is a documentation-only change. Skipping e2e execution."
+            echo "skip-e2e=true" >> $GITHUB_OUTPUT
+          else
+            echo "This is a regular CI build. Continuing ..."
+          fi
+        shell: bash
+
+  e2e:
```

Comment:
Btw. just because you haven't resolved that item here: I added a comment to the configuration stating the reasoning. My understanding was that we don't want to/can't use containers for e2e tests because of the Docker-in-Docker issue. That's why I kept it like that. :thinking: